x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/HistogramFieldMapper.java

```diff
@@ -0,0 +1,462 @@
+                        // should be a number
+                        ensureExpectedToken(XContentParser.Token.VALUE_NUMBER, token, context.parser()::getTokenLocation);
+                        counts.add(context.parser().intValue());
+                        token = context.parser().nextToken();
+                    }
+                } else {
+                    throw new MapperParsingException("error parsing field [" +
+                        name() + "], with unknown parameter [" + fieldName + "]");
+                }
+                token = context.parser().nextToken();
+            }
+            if (values == null) {
+                throw new MapperParsingException("error parsing field ["
+                    + name() + "], expected field called [" + VALUES_FIELD.getPreferredName() + "]");
+            }
+            if (counts == null) {
+                throw new MapperParsingException("error parsing field ["
+                    + name() + "], expected field called [" + COUNTS_FIELD.getPreferredName() + "]");
+            }
+            if (values.size() != counts.size()) {
+                throw new MapperParsingException("error parsing field ["
+                    + name() + "], expected same length from [" + VALUES_FIELD.getPreferredName() +"] and " +
+                    "[" + COUNTS_FIELD.getPreferredName() +"] but got [" + values.size() + " != " + counts.size() +"]");
+            }
+            if (values.size() == 0) {
+                context.path().remove();
+                return;
+            }
+            if (fieldType().hasDocValues()) {
+                BytesStreamOutput streamOutput = new BytesStreamOutput();
+                for (int i = 0; i < values.size(); i++) {
+                    streamOutput.writeDouble(values.get(i));
+                    if (counts.get(i) < 0) {
+                        throw new MapperParsingException("error parsing field ["
+                            + name() + "], ["+ COUNTS_FIELD + "] elements must be >= 0 but got " + counts.get(i));
+                    }
+                    streamOutput.writeVInt(counts.get(i));
+                }
+
+                Field field = new BinaryDocValuesField(simpleName(), streamOutput.bytes().toBytesRef());
+                streamOutput.close();
+                if (context.doc().getByKey(fieldType().name()) != null) {
+                    throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() +
+                        "] doesn't not support indexing multiple values for the same field in the same document");
+                }
+                context.doc().addWithKey(fieldType().name(), field);
+            }
+
+        } catch (Exception ex) {
+            if (ignoreMalformed.value() == false) {
+                throw new MapperParsingException("failed to parse field [{}] of type [{}]",
+                    ex, fieldType().name(), fieldType().typeName());
+            }
```

Comment:
I added my own logic to skip the field to the end. Not sure if there is some other utility to what I am doing here by hand.

Reply:
This is what XContentSubParser has been designed for. See https://github.com/elastic/elasticsearch/pull/35603. Maybe it would be more robust? By the way looking at the latest version of GeoShapeFieldMapper, it looks like it no longer handles ignoreMalformed correctly, or am I misreading it cc @imotov ?